File: 프로그래머스/0/120897. 약수 구하기/약수 구하기.py
import math
def solution(n):
    arr = []
    for i in range(1, math.ceil(math.sqrt(n)) + 1):
            arr += [i, n // i] if n // i != i and n % i == 0 else [i] if n % i == 0 else []
    arr.sort()
    return arr

File: 백준/Bronze/2501. 약수 구하기/약수 구하기.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int k = sc.nextInt();
        int[] arr = new int[(int) Math.floor(Math.sqrt(n)) * 2 + 1];
        int ans = 0;
        int cnt = 0;
        for (int i = 1; i <= n; i++) {
            if (n % i == 0) {
                cnt++;
                arr[cnt] = i;
                if (cnt == k) {
                    ans = i;
                    break;
                }
            }
        }
        System.out.println(ans);
    }
}